crates/hir-ty/src/lower.rs

```diff
@@ -867,15 +887,46 @@ impl<'a> TyLoweringContext<'a> {
                     }
                 }
             }
+
+            for arg in generic_args
+                .args
+                .iter()
+                .filter(|arg| matches!(arg, GenericArg::Lifetime(_)))
+                .skip(skip)
```

Comment:
I think this skip is wrong? It accounts for the self type which is a `GenericArg::TypeParam`, those are already filtered out by the filter at this point though, so we might skip a lifetime here. I don't think the skip is needed at all in this case.

Reply:
It is wrong, I thought to remove it but then I forgot about it.